bundles/framework/admin-announcements/view/AnnouncementsListHandler.js

```diff
@@ -0,0 +1,160 @@
+    const index = existing.indexOf(value);
+    if (index > -1) {
+        existing.splice(index, 1);
+    }
+    // Save back to localStorage
+    localStorage.setItem(name, existing.toString());
+};
+
+class ViewHandler extends StateHandler {
+    constructor (instance) {
+        super();
+        this.instance = instance;
+        this.announcementsHelper = announcementsHelper();
+        this.newTitle = 'Uusi ilmoitus';
+        this.state = {
+            update: false,
+            announcements: [],
+            title: 'Uusi Ilmoitus',
+            updated: false,
+            active: true,
+            activeKey: []
+
+        };
+    }
+
+    getAdminAnnouncements (callback) {
+        this.announcementsHelper.getAdminAnnouncements(function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.getAdminAnnouncementsFailed'));
+                return;
+            }
+            callback(data);
+        });
+    }
+
+    saveAnnouncement (data) {
+        this.announcementsHelper.saveAnnouncement(data, function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.saveFailed'));
+            } else {
+                Messaging.success(getMessage('messages.saveSuccess'));
+            }
+        });
+        this.updateState({
+            activeKey: []
+        });
+    }
+
+    // Update all the announcements f.ex. when saved. Set active key as empty so all panels get closed.
+    updateAnnouncement (data) {
+        this.announcementsHelper.updateAnnouncement(data, function (err, data) {
+            if (err) {
+                Messaging.error(getMessage('messages.updateFailed'));
+                return false;
+            } else {
+                Messaging.success(getMessage('messages.updateSuccess'));
+                removeFromLocalStorageArray('oskari-announcements', data.data[0].toString());
+            }
+        });
+        this.updateState({
+            activeKey: []
+        });
+    }
+
+    deleteAnnouncement (index, id) {
+        this.id = { id };
+        this.announcementsHelper.deleteAnnouncement(this.id, function (err) {
+            if (err) {
+                Messaging.error(getMessage('messages.deleteFailed'));
+            } else {
+                Messaging.success(getMessage('messages.deleteSuccess'));
+            }
+        });
+        const newList = [...this.state.announcements];
+        newList.splice(index, 1);
+
+        // Update accordion with announcements and keep all closed
+        this.updateState({
+            announcements: newList,
+            activeKey: []
+        });
+    }
+    // Cancel creating a new announcement or editing one. Close all panels.
+    cancel (index, id) {
+        if (id !== undefined) {
+            this.updateState({
+                activeKey: []
+            });
+        } else {
+            const newList = [...this.state.announcements];
+            newList.splice(index, 1);
```

Comment:
Do you add a new announcement to the list before it's saved? Or why do you need to splice here?

Reply:
Yes, addForm() pushes a new object into the announcements list when user creates a new announcement (aka panel)